o2checkcode.sh

```diff
@@ -19,20 +19,21 @@ cp "${O2_ROOT}"/compile_commands.json .
 O2_SRC=$(python -c 'import json,sys,os; sys.stdout.write( os.path.commonprefix([ x["file"] for x in json.loads(open("compile_commands.json").read()) if not "G__" in x["file"] and x["file"].endswith(".cxx") ]) )')
 [[ -e "$O2_SRC"/CMakeLists.txt && -d "$O2_SRC"/.git ]]
 
-BASECOMMIT=${O2CHECKCODE_BASECOMMIT:=${ALIBUILD_PR_BASE}}
-HEADCOMMIT=${O2CHECKCODE_HEADCOMMIT:=${ALIBUILD_PR_HEAD}}
-
-# We query Git if a base commit was given, which we treat as an indication for a delta check. With
-# more than 50 files changed we run the checks on all files.
-if [[ $BASECOMMIT && $(cd "$O2_SRC" && git diff $HEADCOMMIT $BASECOMMIT --name-only | wc -l) -le 50 ]]; then
-  O2_CHECKCODE_CHANGEDFILES=$(cd "$O2_SRC" && git diff $HEADCOMMIT $BASECOMMIT --name-only               | \
-                              grep -E '\.cxx$|\.h$'                                                      | \
-                              while read FILE; do [[ -e "$O2_SRC/$FILE" ]] && echo "$FILE" || true; done | \
-                              xargs echo | sed -e 's/ /:/g')
-  if [[ ! $O2_CHECKCODE_CHANGEDFILES ]]; then
-    echo "Nothing changed with respect to base commit: not checking anything"
-    exit 0
-  fi
+# We have something to compare our working directory to (ALIBUILD_UPSTREAM_HASH). We check only the
+# changed files (including the untracked ones) if the list of relevant files that changed is up to
+# 50 entries long
+if [[ $ALIBUILD_UPSTREAM_HASH ]]; then
```

Comment:
How can a user now use the recipe? Does she have to define `ALIBUILD_UPSTREAM_HASH`? In the old version it was quite clear that she had to define more meaningful variables `O2CHECKCODE_...`